### PR TITLE
update tests for the improved query strings obfuscation regex

### PR DESCRIPTION
**Pull request by christophe-papazian**
## Description

Due to changes of the **QueryString - Sensitive Data Obfuscation** RFC, the new improved regex implies some small changes in the redacted result. This PR fixes that.

## Workflow

1. ⚠️⚠️ Create your PR as draft
2. Follow the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
3. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
4. Mark it as ready for review

Once your PR is reviewed, you can merge it! :heart:

## Reviewer checklist

* [ ] Check what scenarios are modified. If needed, add the relevant label (`run-parametric-scenario`, `run-profiling-scenario`...). If this PR modifies any system-tests internal, then add the `run-all-scenarios` label ([more info](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md)). 
* [ ] CI is green
   * [ ] If not, failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] if any of `build-some-image` label is present
  1. is the image labl have been updated ? 
  2. just before merging, locally build and push the image to hub.docker.com
* [ ] if a scenario is added (or removed), add (or remove) it in system-test-dasboard nightly

### CODE REVIEW

**Review by cbeauchesne — CHANGES_REQUESTED** [review]
When a feature is updated, we want to continue tests for the old way for tracers that haven't yet updated their code. The good way to do this is : 


```
@released(tracer_A="1.2", tracer_A="7.3") 
class Test_Feature:
   @irrelevant(tracer_A<"4.0", reason="Tracer A released the new version at 4.0")  
   def test_legacy:
      ...

   @missing_feature(context.library="tracer_B", reason="Tracer B did not yest implemented the new version")  
   def test_new_way:
       ...

File: tests/test_standard_tags.py [hunk 1]
```diff
@@ -57,7 +58,7 @@ def test_url_with_query_string(self):
             value_as_regular_expression=True,
         )
 
-    def setup_url_with_sensitive_query_string(self):
+    def setup_url_with_sensitive_query_string_legacy(self):
```
Comment (simon-id) [inline tests/test_standard_tags.py:61]: if we change the name, won't the manifest files all need to be updated ?

Comment (christophe-papazian) [reply tests/test_standard_tags.py:61]: no problem here, as we don't change class name.

File: tests/test_standard_tags.py [hunk 2]
```diff
@@ -80,19 +80,71 @@ def setup_url_with_sensitive_query_string(self):
             ),
         ]
 
+    # when tracer is updated, add (for exemple)
```
Comment (simon-id) [inline tests/test_standard_tags.py:83]: `exemple` is french
```suggestion
    # when tracer is updated, add (for example)
```

File: tests/test_standard_tags.py [hunk 3]
```diff
@@ -80,19 +80,71 @@ def setup_url_with_sensitive_query_string(self):
...
+    # @irrelevant(context.library >= "python@1.20.0", reason="python released the new version at 1.20.0")
     def test_url_with_sensitive_query_string(self):
```
Comment (simon-id) [inline tests/test_standard_tags.py:85]: two tests have the same name now ?? line 85 and 118. I think you mean this:
```suggestion
    def test_url_with_sensitive_query_string_legacy(self):
```

**Review by simon-id — APPROVED** [review]

**Pull request merged**